pandas/tests/indexing/multiindex/test_loc.py

```diff
@@ -764,6 +764,25 @@ def test_loc_getitem_index_differently_ordered_slice_none():
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize("indexer", [[1, 2, 7, 6, 2, 3, 8, 7], [1, 2, 7, 6, 3, 8]])
+def test_loc_getitem_index_differently_ordered_slice_none_duplicates(indexer):
+    # GH#40978
+    df = DataFrame(
+        [1] * 8,
+        index=MultiIndex.from_tuples(
+            [(1, 1), (1, 2), (1, 7), (1, 6), (2, 2), (2, 3), (2, 8), (2, 7)]
+        ),
+        columns=["a"],
+    )
+    result = df.loc[(slice(None), indexer), :]
```

Comment:
can you add this as an equivalent test: `df_sliced = df.loc[df.index.isin(all_items, level=1), :]` should yield the same expected

Reply:
Added test, but should not return the same since isin produces an boolean mask, which does not reorder the underlying dataframe. This should be the identity